app/controllers/api/stories_controller.rb

```diff
@@ -11,16 +11,26 @@ def random
     show
   end
 
-  def resource
-    @story ||= if params[:id]
-                 Story.published.visible.find(params[:id])
-               elsif request.put? || request.post?
-                 Story.new
-               end
+  private
+
+  def resources_base
+    if highlighted?
+      account.portfolio_stories
+    else
+      super
+    end
   end
 
-  def resources
-    @stories ||= resources_base.includes(
+  def filtered(resources)
+    if highlighted?
+      resources
+    else
+      super
+    end
+  end
+
+  def included(relation)
```

Comment:
I thought that the show cache key was pegged on the resource, which is only fetched once, is that not the case?

I can exclude it for `show` but it seems like it's sort of a toss up which is better. Roar is by far the most expensive part of our controller actions.